doc/reference/trd-appid.md

```diff
@@ -0,0 +1,795 @@
+policies. 
+
+For example, suppose that a device wants to grant access to
+all Userspace Binaries signed by a certain 3072-bit RSA key K and has
+no other security policies. The Credentials Checking Policy only
+accepts `Rsa3072KeyWithID` credentials with key K. The `Compress`
+trait implementation assigns a Short ID as the ID stored in the
+credentials. Access control systems within the kernel can define their
+policies in terms of these identifiers, such that they can check
+access by comparing 32-bit integers rather than 384-byte keys.
+
+The 32-bit value MUST be non-zero. `ShortID` uses `core::num::NonZeroU32`
+so that an `Option<ShortID>` can be 32 bits in size, with 0 reserved
+for `None`.
+
+```rust
+#[derive(Clone, Copy, Eq)]
+struct ShortID {
+    id: core::num::NonZeroU32
+}
+
+pub trait Compress {
+    fn to_short_id(process: &dyn Process) -> Option<ShortID>;
+}
+```
+
+The `to_short_id` method returns an `Option` so that it has a clear
+default action if the Process Checker does not wish to assign it any
+meaningful identifier that might allow additional access.  A return
+value of `None` semantically means that the process's Application
+Credentials or other state do not map to any known security group or
+set of privileges. A`Some` result means the credentials map to a known
+security group or set of privileges.
+
+Generally, the Process Checker that implements `AppCredentialsChecker`
+and `AppIdentification` also implements `Compress`. This allows it to
+share copies of public keys or other credentials that it uses to make
+decisions, reducing flash space dedicated to these constants. Doing so
+also makes it less likely that the two are inconsistent.
+ 
+`ShortID` values MUST be locally unique among running processes.  The
+mapping between Global Application Identifiers and `ShortID` values
+MUST be deterministic.  Kernels SHOULD implement `Compress` in a
+manner that minimizes the chance that two different Application
+Identifiers compress to the same Short ID (e.g., taking the low-order
+bits of a strong cryptographic hash function, or using a known,
+deterministic mapping).
+
+Short IDs are locally unique for three reasons. First, it simplifies
+process management and naming: a particular application identifier
+uniquely identifies a running process. Second, it ensures that
+resources bound to an application identifier (such as non-volatile
+storage) do not have to handle concurrent accesses from multiple
+processes. Finally, generally one does not want two copies of the same
+Application running: they can create conflicting responses and
+behaviors.
+
+For example, suppose there is a system that wants to grant extra
+permissions to Tock binaries with a `TbfFooterV2Credentials` of
+`Rsa4096Key` with a certain public key. The public key is the Global
+Application Identifier of the process binary. Note this means only one
+process signed with that key can run at any time.
```

Comment:
Yeah, I think it would make sense to rephrase this. The use case is that there is a one-to-one key-to-app mapping. The fact that it's signed with a key is what establishes it is that specific app.

Reply:
Rephrased.